### PR TITLE
MINOR: Improve ometa logging

### PR DESCRIPTION
**Pull request by ulixius9**
# Add JWT Token Decoding and User Logging for OpenMetadata Authentication

## Summary
This PR adds JWT token decoding functionality to log authenticated user information when using OpenMetadata authentication provider. The feature helps with debugging and monitoring authentication flows by extracting and logging user details from JWT tokens.

### 1. Added JWT Token Decoding Utility (`ingestion/src/metadata/ingestion/ometa/utils.py`)
- **New function**: `decode_jwt_token(jwt_token: str) -> Optional[Dict[str, Any]]`
- Safely decodes JWT token payload without verification using built-in Python libraries
- Handles base64 URL-safe decoding with proper padding
- Extracts JSON payload from JWT tokens
- Includes comprehensive error handling and debug logging

### 2. Enhanced OpenMetadata Client Initialization (`ingestion/src/metadata/ingestion/ometa/ometa_api.py`)
- **Added JWT token processing** in the `__init__` method
- **Conditional logging**: Only processes JWT tokens when `authProvider` is set to "openmetadata"
- **User extraction**: Attempts to extract user name from multiple JWT claim fields:
  - `sub` (standard JWT subject claim)
- **Debug logging**: Logs authenticated user name at debug level
- **Error resilience**: Gracefully handles malformed tokens or decoding errors

### 3. Comprehensive Test Coverage (`ingestion/tests/unit/test_ometa_utils.py`)
- **9 new test cases** covering various scenarios:
  - Valid JWT token decoding
  - Base64 padding handling
  - Invalid token formats
  - Invalid base64 encoding
  - Invalid JSON payload
  - Empty payloads
  - None/empty input handling
  - Real-world OpenMetadata JWT token examples

### JWT Token Structure
The implementation handles standard JWT tokens with the format: `header.payload.signature`
- Only decodes the payload (second part) for user information extraction
- Uses base64 URL-safe decoding with automatic padding
- No cryptographic verification (for debugging purposes only)

### User Name Extraction Logic
The system attempts to extract user names from JWT claims in the following order:
1. `sub` claim (standard JWT subject)
2. `username` claim (custom field)
3. `name` claim (display name)
4. Username portion of `email` claim (fallback)

### Error Handling
- Invalid JWT format (wrong number of parts)
- Invalid base64 encoding
- Invalid JSON payload
- Missing or malformed claims
- All errors are logged at debug level without affecting normal operation

## Benefits
1. **Enhanced Debugging**: Provides visibility into which user is authenticating
2. **Monitoring**: Helps track authentication patterns and user activity
3. **Troubleshooting**: Assists in resolving authentication-related issues
4. **No External Dependencies**: Uses only built-in Python libraries
5. **Safe Implementation**: No impact on existing functionality, debug-level logging only

## Testing
- All new functionality is covered by comprehensive unit tests
- Tests include edge cases and error scenarios
- Real-world JWT token examples are tested
- No breaking changes to existing functionality

## Backward Compatibility
- ✅ No breaking changes
- ✅ Optional feature (only active when authProvider is "openmetadata")
- ✅ Debug-level logging (doesn't affect production logs)
- ✅ Graceful error handling (doesn't interrupt normal operation)

## Example Usage
When initializing OpenMetadata client with JWT authentication:
```python
config = OpenMetadataConnection(
    hostPort="http://localhost:8585/api",
    authProvider=AuthProvider.openmetadata,
    securityConfig=OpenMetadataJWTClientConfig(jwtToken="your-jwt-token")
)
metadata = OpenMetadata(config)
# Debug log will show: "Authenticated user: username"
```

This enhancement provides valuable debugging information while maintaining the robustness and reliability of the OpenMetadata authentication system.

### CODE REVIEW

File: ingestion/src/metadata/ingestion/ometa/ometa_api.py [hunk 1]
```diff
@@ -157,6 +162,25 @@ def __init__(
 
         self._auth_provider = OpenMetadataAuthenticationProvider.create(self.config)
 
+        # Log user name from JWT token if authProvider is openmetadata
+        if (
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/ometa/ometa_api.py]: i'd move this chunk out to a method on its own, instead of keeping it inside the init

File: ingestion/src/metadata/utils/secrets/aws_secrets_manager.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ def get_string_value(self, secret_id: str) -> Optional[str]:
         try:
             kwargs = {"SecretId": secret_id}
             response = self.client.get_secret_value(**kwargs)
-            logger.debug("Got value for secret %s.", secret_id)
+            logger.info("Got value for secret %s.", secret_id)
```
Comment (pmbrull) [inline ingestion/src/metadata/utils/secrets/aws_secrets_manager.py]: instead of just changing the logging here, could we have it at the ABC level? so we know all implementations will be logged correctly

Comment (pmbrull) [reply ingestion/src/metadata/utils/secrets/aws_secrets_manager.py]: or on the custom_pydantic side, when we have the `_get_secret` (however that's called)

**Review by pmbrull — APPROVED** [review]
